people-and-planet-ai/land-cover-classification/trainer/tf_model.py

```diff
@@ -0,0 +1,239 @@
+    inputs = tf.io.parse_tensor(example["inputs"], tf.float32)
+    labels = tf.io.parse_tensor(example["labels"], tf.uint8)
+
+    # TensorFlow cannot infer the shape's rank, so we set the shapes explicitly.
+    inputs.set_shape([None, None, NUM_INPUTS])
+    labels.set_shape([None, None, 1])
+
+    # Classifications are measured against one-hot encoded vectors.
+    one_hot_labels = tf.one_hot(labels[:, :, 0], NUM_CLASSES)
+    return (inputs, one_hot_labels)
+
+
+def read_dataset(data_path: str) -> tf.data.Dataset:
+    """Reads compressed TFRecord files from a directory into a tf.data.Dataset.
+
+    Args:
+        data_path: Local or Cloud Storage directory path where the TFRecord files are.
+
+    Returns: A tf.data.Dataset with the contents of the TFRecord files.
+    """
+    file_pattern = tf.io.gfile.join(data_path, "*.tfrecord.gz")
+    file_names = tf.data.Dataset.list_files(file_pattern).cache()
+    dataset = tf.data.TFRecordDataset(file_names, compression_type="GZIP")
+    return dataset.map(read_example, num_parallel_calls=tf.data.AUTOTUNE)
+
+
+def split_dataset(
+    dataset: tf.data.Dataset,
+    batch_size: int = BATCH_SIZE,
+    train_test_ratio: int = TRAIN_TEST_RATIO,
+) -> Tuple[tf.data.Dataset, tf.data.Dataset]:
+    """Splits a dataset into training and validation subsets.
+
+    Args:
+        dataset: Full dataset with all the training examples.
+        batch_size: Number of examples per training batch.
+        train_test_ratio: Percent of the data to use for training.
+
+    Returns: A (training, validation) dataset pair.
+    """
+    # For more information on how to optimize your tf.data.Dataset, see:
+    #   https://www.tensorflow.org/guide/data_performance
+    indexed_dataset = dataset.enumerate()  # add an index to each example
+    train_dataset = (
+        indexed_dataset.filter(lambda i, _: i % 100 <= train_test_ratio)
+        .map(lambda _, data: data, num_parallel_calls=tf.data.AUTOTUNE)  # remove index
+        .cache()  # cache the individual parsed examples
+        .shuffle(SHUFFLE_BUFFER_SIZE)  # randomize the examples for the batches
+        .batch(batch_size)  # batch randomized examples
+        .prefetch(tf.data.AUTOTUNE)  # prefetch the next batch
+    )
+    validation_dataset = (
+        indexed_dataset.filter(lambda i, _: i % 100 > train_test_ratio)
+        .map(lambda _, data: data, num_parallel_calls=tf.data.AUTOTUNE)  # remove index
+        .batch(batch_size)  # batch the parsed examples, no need to shuffle
+        .cache()  # cache the batches of examples
+        .prefetch(tf.data.AUTOTUNE)  # prefetch the next batch
+    )
+    return (train_dataset, validation_dataset)
+
+
+def create_model(
+    dataset: tf.data.Dataset, kernel_size: int = KERNEL_SIZE
+) -> tf.keras.Model:
+    """Creates a Fully Convolutional Network Keras model.
+
+    Make sure you pass the *training* dataset, not the validation or full dataset.
```

Comment:
This _seems_ like a big warning, do you think there'd be any ways you could add one extra layer of security? Not sure how this could be done, but I'd imagine some users would accidentally use the wrong dataset and get upset.

If this is mentioned in other parts (i.e. README / Notebooks) that I missed, please feel free to ignore this comment :)

Reply:
Sounds good.